notebook/gateway/managers.py

```diff
@@ -17,37 +17,47 @@
 
 from traitlets import Instance, Unicode, Float, Bool, default, validate, TraitError
 from traitlets.config import SingletonConfigurable
+from ..db_util import ExecuteQueries
 
 
 class GatewayClient(SingletonConfigurable):
-    """This class manages the configuration.  It's its own singleton class so that we
-       can share these values across all objects.  It also contains some helper methods
-        to build request arguments out of the various config options.
-
+    """
+    This class manages the configuration. It's its own singleton class so that we can share these values across
+    all objects. It also contains some helper methods to build request arguments out of the various config options.
     """
 
-    url = Unicode(default_value=None, allow_none=True, config=True,
+    urls = Unicode(default_value=None, allow_none=True, config=True,
         help="""The url of the Kernel or Enterprise Gateway server where
         kernel specifications are defined and kernel management takes place.
         If defined, this Notebook server acts as a proxy for all kernel
         management and kernel specification retrieval.  (JUPYTER_GATEWAY_URL env var)
         """
     )
 
-    url_env = 'JUPYTER_GATEWAY_URL'
+    ml_nodes = ExecuteQueries().get_mlnodes()
+    ml_node_urls = []
+    for record in ml_nodes:
+        ml_node_urls.append(str(record.ip_address) + ':8888')
+
+    urls = ml_node_urls
+
+    urls_env = 'JUPYTER_GATEWAY_URL'
 
-    @default('url')
+    @default('urls')
     def _url_default(self):
         return os.environ.get(self.url_env)
 
-    @validate('url')
+    @validate('urls')
     def _url_validate(self, proposal):
```

Comment:
think, you should add comments to all the methods

Reply:
I haven't written those function in which the comments are not there. I have added to the ones that I have updated.

Reply:
Added a comment.